build-package.sh

```diff
@@ -273,10 +273,17 @@ source "$TERMUX_SCRIPTDIR/scripts/build/termux_step_finish_build.sh"
 if [ "$TERMUX_ON_DEVICE_BUILD" = "true" ]; then
 	# For on device builds cross compiling is not supported.
 	# Target architecture must be same as for environment used currently.
-	TERMUX_ARCH=$(dpkg --print-architecture 2>/dev/null || pacman-conf | grep Architecture | sed 's/Architecture = //g')
+	case "$TERMUX_MAIN_PACKAGE_FORMAT" in
+		"debian") TERMUX_ARCH=$(dpkg --print-architecture);;
+		"pacman") TERMUX_ARCH=$(pacman-conf | grep Architecture | sed 's/Architecture = //g');;
+	esac
 	export TERMUX_ARCH
 fi
 
+if [ -n "$TERMUX_MAIN_PACKAGE_FORMAT" ]; then
+	TERMUX_PACKAGE_FORMAT="$TERMUX_MAIN_PACKAGE_FORMAT"
```

Comment:
Why is two parameters, TERMUX_PACKAGE_FORMAT and TERMUX_MAIN_PACKAGE_FORMAT, with the same value needed? 

What's the difference between the two parameters?

Reply:
TERMUX_PACKAGE_FORMAT - this value specifies which package format will be compiled
TERMUX_MAIN_PACKAGE_FORMAT - this is a value that indicates which basic packet format the device is using  

This change will allow the device to compile the recommended package format, that is, the format that the device uses.

Reply:
Ok, thanks for explaining, seems reasonable!

What is 
```
if [ -n "$TERMUX_MAIN_PACKAGE_FORMAT" ]; then
	TERMUX_PACKAGE_FORMAT="$TERMUX_MAIN_PACKAGE_FORMAT"
fi
```
suppose to do though? Right now it overrides TERMUX_PACKAGE_FORMAT if TERMUX_MAIN_PACKAGE_FORMAT is set. Seems like that would break the `--format` flag

Reply:
Please explain what the code is suppose to do. 

To me it seems like it will override the --format arg given by user.

Reply:
First, the value of TERMUX_PACKAGE_FORMAT is set, then after that the --format flag is checked (and the value changes even if it has already been set).

Reply:
Alright, please change so that it defaults to TERMUX_PACKAGE_MAIN_FORMAT for on device builds in [termux_step_setup_variables](https://github.com/termux/termux-packages/blob/master/scripts/build/termux_step_setup_variables.sh#L2) instead, to gather (most) of the defaults values there

Reply:
Check, please